pkg/gui/controllers/helpers/inline_status_helper.go

```diff
@@ -61,18 +64,33 @@ func (self inlineStatusHelperTask) Continue() {
 }
 
 func (self *InlineStatusHelper) WithInlineStatus(opts InlineStatusOpts, f func(gocui.Task) error) {
-	self.c.OnWorker(func(task gocui.Task) {
-		self.start(opts)
-
-		err := f(inlineStatusHelperTask{task, self, opts})
-		if err != nil {
-			self.c.OnUIThread(func() error {
-				return self.c.Error(err)
-			})
-		}
+	context := self.c.ContextForKey(opts.ContextKey)
+	view := context.GetView()
+	visible := view.Visible && self.windowHelper.TopViewInWindow(context.GetWindowName()) == view
+	if visible {
+		self.c.OnWorker(func(task gocui.Task) {
+			self.start(opts)
+
+			err := f(inlineStatusHelperTask{task, self, opts})
+			if err != nil {
+				self.c.OnUIThread(func() error {
+					return self.c.Error(err)
+				})
+			}
 
-		self.stop(opts)
-	})
+			self.stop(opts)
+		})
+	} else {
+		message := presentation.ItemOperationToString(opts.Operation, self.c.Tr)
```

Comment:
Is it ok to have helpers depend on things in presentation? It feels kind of wrong to me, but I wasn't sure. Should I move the ItemOperationToString function somewhere else to avoid this?